app/scripts/ui.js

```diff
@@ -117,11 +117,15 @@ async function start() {
   if (isManifestV3) {
     /*
      * In case of MV3 the issue of blank screen was very frequent, it is caused by UI initialising before background is ready to send state.
-     * Code below ensures that UI is rendered only after CONNECTION_READY message is received thus background is ready.
+     * Code below ensures that UI is rendered only after "CONNECTION_READY" or "startUISync"
+     * messages are received thus the background is ready, and ensures that streams and
+     * phishing warning page load only after the "startUISync" message is received.
      * In case the UI is already rendered, only update the streams.
      */
     const messageListener = async (message) => {
-      if (message?.name === 'CONNECTION_READY') {
+      if (message?.name === 'CONNECTION_READY' && !isUIInitialised) {
+        initializeUiWithTab(activeTab);
```

Comment:
This seems wrong. It looks like  `CONNECTION_READY` can be sent before the user is logged in, which would result in state being sent before the keyrings are initialized.

Reply:
I see that this was added here: 1710cbec36360fb649d6fb683b9751986295dce0

Maybe this would be alleviated by this PR? https://github.com/MetaMask/metamask-extension/pull/16559